### ISSUE TITLE
VL neu: Java/JVM: Collections

### ISSUE THREAD

**Issue by cagix** [PM-Dungeon/PM-Lecture#243]

**Comment by jedi101** [comment]
@AMatutat - Gibt es hier vielleicht schon Inhalt auf den ich mich stürzen könnte für das Quiz?

**Comment by AMatutat** [comment]
@jedi101 ne, noch nicht.

**Comment by jedi101** [comment]
@AMatutat, alles klar. Dann kümmere ich mich erst um das Quiz mit den Exceptiones.

**Comment by cagix** [comment]
@AMatutat @jedi101 Neee, hier gibts leider noch nix. Und mit Blick auf meinen Kalender und die Tatsache, dass ich heute noch die zweite von drei Threading-Sachen fertig kriege (hoffentlich) und morgen die dritte und Samstag dann Exceptions: Das wird nix mehr für nächste Woche :/ 

Was haltet ihr davon? Ich würde das ne Woche nach hinten schieben und dafür das Framework-Thema (das ist nämlich fertig) auf nächste Woche?

**Comment by cagix** [comment]
> @cagix hast du bestimmte Vorstellungen für das Thema?

Huch? Das hat sich grad überkreuzt ... Nein, ich habe keine Vorstellungen. Ich hatte mir mal https://dev.java/learn/the-collections-framework/ als gute Referenz notiert. Nur das ist nix, was ich am Sonntag auf einer Pobacke abrocke inkl. eigener Ideen und Beispiele ... das braucht länger.

**Comment by AMatutat** [comment]
> Was haltet ihr davon? Ich würde das ne Woche nach hinten schieben und dafür das Framework-Thema (das ist nämlich fertig) auf nächste Woche?

Die labels verwirren mich immer noch. Aber klingt doch nach einen Plan.

**Comment by AMatutat** [comment]
Inhalt Buleltpoints:
- Collection und/vs class Collections. (die Benennung ist halt echt wieder typisch java.... Dachte Klassen sind immer singular.....)
- Collection Framework
  - mit den wichtigsten Methoden 
  - Hierarchy
- Die "wichtigsten" Collection
  - List
    - Unterschiedliche Sub-Typen und warum ArrayList nicht immer das beste ist
   - Set/Sorted Set
     - HashSet und how to hash
   - Queue/ Dequeue
 - Iterator Interface (@cagix oder hattest du das schonmal irgendwo)
 -  (class) Collections Methoden 



Sehe ich das richtig, `Map` gehört nicht zu Collection?

**Comment by AMatutat** [comment]
Klingt gut.

**Comment by AMatutat** [comment]
Würde es Sinn machen Map in einer eigenen Vorlesung zu machen?
Die wäre natürlich entsprechend kurz aber eigentlich ist das ja gar nicht so schlecht und zusätzlich trennt man das ganze nochmal etwas von collections ab.

**Comment by cagix** [comment]
> Würde es Sinn machen Map in einer eigenen Vorlesung zu machen?
> Die wäre natürlich entsprechend kurz aber eigentlich ist das ja gar nicht so schlecht und zusätzlich trennt man das ganze nochmal etwas von collections ab. 

nee, dass gehört schon mit dazu. ich würde aber nur ein uml mit den wichtigsten klassen malen, der rest ist notes (wichtigste unterschiede).
